assets/src/design-system/components/contextMenu/menu-item.js

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useEffect, useMemo, useRef } from 'react';
+import styled, { css } from 'styled-components';
+/**
+ * Internal dependencies
+ */
+import { THEME_CONSTANTS } from '../../theme';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '../button';
+import { Link } from '../typography/link';
+import { Text } from '../typography/text';
+
+const ItemText = styled(Text)`
+  width: 200px;
+  text-align: left;
+`;
+
+const Shortcut = styled(Text)(
+  ({ theme }) => css`
+    color: ${theme.colors.border.disable};
+    white-space: nowrap;
+  `
+);
+
+export const MenuItem = ({
+  disabled,
+  focusedIndex,
+  href,
+  index,
+  label,
+  onClick,
+  setFocusedIndex,
+  shortcut,
+}) => {
+  const itemRef = useRef(null);
+
+  const handleFocus = () => {
+    if (focusedIndex !== index) {
+      setFocusedIndex(index);
+    }
+  };
```

Comment:
Keeps focus functionality consistent even if we are interchanging between `tab` and arrow keys.